### ISSUE TITLE
Split transformCounts to row and col specific functions

### ISSUE THREAD

**Issue by antagomir** [microbiome/mia#93]
The current `transformCounts` contains operations for both samples (clr, hellinger etc.) and features (Z). In the future, we may like to add some that could be applied in either direction (rank index).

1. Splitting `transformCounts` into `transformSamples` and `transformFeatures` would be a more clear and informative naming scheme.

2. Can we consider changing the argument name "abund_values" into "assay" since this is what it essentially does - conceptually more clear?

**Comment by FelixErnst** [comment]
My 2 cents:

1. I think the was discussed priviously, wasn't it? I think it would make much more sense. However, note that currently only Z transformation works in a rowwise manner and is not availavble through `transformCounts`. So the split is effectively alread in place. (`transformCounts` would need to be kept as an alias form `transformSamples` for backwards compatibility)
2. Why? All functions use either `abund_values` or `exprs_values` throughout multiple packages. In addition, the argument name `assay` is not clear enough, since for a new user it would suggest the assay values themselves to be supplied to such a function. `assay_name` would be better.
I am also not sure what is conceptually better by using `assay(_name)`. The term `assay` is my mind a technical term, whereas `abund_values` is a conceptual name. So I am not sure what you mean exactly. What would improve? The technical understanding from a users point of view?

**Comment by antagomir** [comment]
1. Yes we discussed this earlier but it has not been implemented while we got more experience with incorporating these things in workflows. It seems that we agree, so we could proceed with this split? 
2. Right, for compatibility with other pkgs perhaps it is justified then indeed. For me it is easier to thinkg of which "assay" we are operating on (as these are assays anyway) and abund_values does not explicitly refer to that. But it is ok.

**Comment by FelixErnst** [comment]
Sure go ahead with the implementation. I think this just a bit of rearranging and generalization need. I think the current argument name has its merrits from the conceptual side of things

**Comment by antagomir** [comment]
Great. Me or @tvborm will do this when time allows.

**Comment by antagomir** [comment]
Very good that this is now implemented.

I just noticed that examples include ecological (sample-wise) transformations, being applied to features. 


```
# If you want to do the transformation for features, you can do that by using      
transformFeatures(x, method="hellinger", name="hellinger_features")
head(assay(x, "hellinger_features"))
```

This may have missed my attention previously.

Many of the transformations are ecologically motivated, and essentially never done for features. I propose to disable the use of such transformations for features, and update the examples accordingly.

More specifically,

Both `transformSamples` and `transformFeatures` can include 
- log10
- pa

`transformSamples` can include 
- rank
- relabundance
- hellinger
- clr

`transformFeatures` can include 
- Z

**Comment by TuomasBorman** [comment]
Ok, I can do that, it shouldn't be big thing. The intention was to allow all kind of transformations. I didn't really think about the thing from this point of view

**Comment by antagomir** [comment]
Ok great! We are helping the users by not allowing them to make very unconventional (and probably misinformed) choices.

**Comment by FelixErnst** [comment]
Sorry I missed this, but this can be easily fixed.

Just remove the offending options from the generic and method and include a match call to `match.arg`, e.g. `method <- match.arg(method, c("clr", "hellinger", "log10", "pa", "rank", "relabundance"))` for transformSamples.

What I missed as well is the addition of `transformSamples` generic and method. `transformCounts` should be only a wrapper for `transformSamples` as discussed above

**Comment by antagomir** [comment]
Seems good overall but Felix may better comment the generics and methods.

**Comment by FelixErnst** [comment]
Closed via #102
